src/main/resources/avro/bdg.avdl

```diff
@@ -955,51 +670,199 @@ record VariantAnnotation {
-   The values are stored as strings, even for flag, integer, and float types.  VCF
+   The values are stored as strings, even for flag, integer, and float types. VCF
    INFO key values with Number=., Number=0, Number=1, and Number=[n] are shared across
-   all alleles in the same VCF record. VCF INFO key values with Number=A and Number=R
-   are split for multi-allelic sites.
+   all alternate alleles in the same VCF record. VCF INFO key values with Number=A and
+   Number=R are split for multi-allelic sites.
+   */
+  map<string> attributes = {};
+}
+
+/**
+ Allele encodings for genotypes.
+ */
+enum Allele {
+
+  /**
+   Reference allele.
+   */
+  REF,
+
+  /**
+   Alternate allele.
+   */
+  ALT,
+
+  /**
+   An unspecified other alternate allele. This may occur when the variant for
+   the genotype partially encoded by this allele is split as a multi-allelic site.
+   */
+  OTHER_ALT,
+
+  /**
+   Missing allele.
+   */
+  NO_CALL
+}
+
+/**
+ Genotype.
+ */
+record Genotype {
+
+  /**
+   The variant for this genotype.
+   */
+  union { null, Variant } variant = null;
+
+  /**
+   The reference contig the variant for this genotype exists on. VCF column 1 "CONTIG".
+   Used when the variant field has been left unset or set to null for performance reasons.
+   */
+  union { null, string } contigName = null;
+
+  /**
+   The zero-based start position of the variant for this genotype on the reference contig.
+   VCF column 2 "POS" converted to zero-based coordinate system, closed-open intervals.
+   Used when the variant field has been left unset or set to null for performance reasons.
+   */
+  union { null, long } start = null;
+
+  /**
+   The zero-based, exclusive end position of the variant for this genotype on the reference
+   contig. Calculated by Variant.start + Variant.referenceAllele.length(). Used when the
+   variant field has been left unset or set to null for performance reasons.
+   */
+  union { null, long } end = null;
+
+  /**
+   Sample identifier for this genotype. Join with Sample.sampleId for sample metadata.
+   VCF header line sample id.
+   */
+  union { null, string } sampleId = null;
+
+  /**
+   One or more allele values encoding this genotype. If this genotype is phased, the order of
+   the allele values is significant. VCF genotype field reserved key GT mapped to Allele enum
+   values.
+   */
+  array<Allele> alleles = [];
+
+  /**
+   True if this genotype is phased. If true, the order of alleles is significant. VCF genotype field
+   reserved key GT allele separators: '|' genotype is phased; '/' genotype is unphased.
+   */
+  union { boolean, null } phased = false;
+
+  /**
+   Phase set identifier, if any. Phased genotypes on the same contig with the same phaseSetId
+   are in the same phased set. All phased genotypes that do not have a phaseSetId are assumed to
+   belong to the same phased set. VCF genotype field reserved key PS.
+   */
+  union { null, string } phaseSetId = null;
```

Comment:
I think I'd prefer to stick with Int, as that's what the VCF spec states, and what is IMO correct (i.e., if I were to implement this, I would use an Int "UUID"). That being said, I agree that it is pretty common to see the string phase set ID. I don't know what's the correct thing to do here. If we go with the string, then we emit headers that disagree with the VCF spec, which seems wrong. If we go with Int, then we can't handle VCFs that come in from the wild. If we went with say:

```
union { null, int } phaseSetId = null;
union { null, string } phaseSetStringId = null;
```

Then we'd need to figure out at save whether the user provided string/int/no phase set IDs.

I don't think there's a right/wrong answer, but we see this elsewhere (https://github.com/bigdatagenomics/adam/issues/1213), so we should hash out our philosophy.